libraries/AP_Scheduler/PerfInfo.cpp

```diff
@@ -29,6 +29,42 @@ void AP::PerfInfo::ignore_this_loop()
     ignore_loop = true;
 }
 
+// allocate the task info array
+void AP::PerfInfo::allocate_task_info(uint8_t num_tasks)
+{
+    task_info = new TaskInfo[num_tasks];
+    if (task_info == nullptr) {
+        hal.console->printf("Unable to allocate scheduler TaskInfo\n");
+    }
+}
+
+// free the task info array
+void AP::PerfInfo::free_task_info()
+{
+    delete[] task_info;
+    task_info = nullptr;
+}
+
+// update task info
+void AP::PerfInfo::update_task_info(uint8_t task_index, uint16_t task_time_us, bool overrun)
+{
+    if (task_info == nullptr) {
+        return;
+    }
+
+    TaskInfo& ti = task_info[task_index];
+    ti.max_time_us = MAX(ti.max_time_us, task_time_us);
+    if (ti.min_time_us <= 0) {
```

Comment:
It's unlikely to be less than zero given it is unsigned.

Reply:
Fixed